### ISSUE TITLE
Where did you stop and why?

### ISSUE THREAD

**Issue by qpwo** [alexflint/logical-induction#1]
Do you think speed could be incrementally improved to the point of being practically useful?

**Comment by alexflint** [comment]
Well yes I think so. Happy to discuss!

**Comment by qpwo** [comment]
I'll have to dig into it but first thing comes to mind is using a JIT interpreter would be a pretty large speedup. Which of the exponents if any in the algorithm itself do you think could be reduced?

**Comment by alexflint** [comment]
Right! The algorithm can look through traders for a long time before it
finds one that characterizes any meaningful patterns in the current market,
so a big improvement would be to more intelligently look for traders that
exploit the current market.
